objectRecognitionSource/src/test/java/io/vantiq/extsrc/objectRecognition/neuralNet/TestNoProcessor.java

```diff
@@ -0,0 +1,502 @@
+            lastFilename = "objectRecognition/" + SOURCE_NAME + '/' + npProcessor.lastFilename;
+            vantiqResponse = vantiq.selectOne("system.documents", lastFilename);
+            if (vantiqResponse.hasErrors()) {
+                List<VantiqError> errors = vantiqResponse.getErrors();
+                for (int i = 0; i < errors.size(); i++) {
+                    if (errors.get(i).getCode().equals(NOT_FOUND_CODE)) {
+                        fail();
+                    }
+                }
+            }
+            vantiqSavedFiles.add(lastFilename);
+
+        } finally {
+            // delete the directory even if the test fails
+            if (d.exists()) {
+                deleteDirectory(OUTPUT_DIR);
+            }
+
+            npProcessor.close();
+        }
+    }
+
+    // Similar to testImageSavingLocal() and testImageSavingBoth(), but saveImage is set to "vantiq".
+    // No images should be saved locally.
+    @Test
+    public void testImageSavingVantiq() throws ImageProcessingException, InterruptedException {
+
+        // Only run test with intended vantiq availability
+        assumeTrue(testAuthToken != null && testVantiqServer != null);
+
+        String lastFilename;
+        Map config = new LinkedHashMap<>();
+        NoProcessor npProcessor = new NoProcessor();
+
+        config.put("outputDir", OUTPUT_DIR);
+        config.put("saveRate", SAVE_RATE);
+        config.put("saveImage", "vantiq");
+        npProcessor.setupImageProcessing(config, SOURCE_NAME, MODEL_DIRECTORY, testAuthToken, testVantiqServer);
+
+        File d = new File(OUTPUT_DIR);
+        try {
+            // Ensure no results from previous tests
+            if (d.exists()) {
+                deleteDirectory(OUTPUT_DIR);
+            }
+
+            NeuralNetResults results = npProcessor.processImage(getTestImage());
+            assert results == null;
+
+            // Should not exist since images are not being saved locally.
+            assert !d.exists();
+
+            // Checking that image was saved to VANTIQ
+            Thread.sleep(1000);
+            lastFilename = "objectRecognition/" + SOURCE_NAME + '/' + npProcessor.lastFilename;
+            vantiqResponse = vantiq.selectOne("system.documents", lastFilename);
+            if (vantiqResponse.hasErrors()) {
+                List<VantiqError> errors = vantiqResponse.getErrors();
+                for (int i = 0; i < errors.size(); i++) {
+                    if (errors.get(i).getCode().equals(NOT_FOUND_CODE)) {
+                        fail();
+                    }
+                }
+            }
+            vantiqSavedFiles.add(lastFilename);
+
+        } finally {
+            // delete the directory even if the test fails
+            if (d.exists()) {
+                deleteDirectory(OUTPUT_DIR);
+            }
+
+            npProcessor.close();
+        }
+    }
+
+    @Test
+    public void testQuery() throws ImageProcessingException, InterruptedException {
```

Comment:
If I'm reading this correctly, this is testing the internal saving.  I think we need something that verifies that the overall source handles queries correctly when no-processing is selected.  That is, if I write a VAIL procedure that does a select against this source (which can be used, IIRC, to poll the camera, we need to ensure that the select doesn't time out/error.  It should get back a no-content which will be (correctly) interpreted as "no rows" I think.  I'm concerned because the section above seemed to simply skip returning with there was no result.   Generally, that's not good because the overall protocol from the server side is to send the query then await the result.  Failure to get it times out, but it takes a while.  And that's now how the source should behave.

(If I've misread things, please correct me!)